### PR TITLE
Add notice about `cupy.array_api` removal

### LINKED ISSUE #8470: Clarify the future of the `cupy.array_api ` submodule and the `array_api_compat.cupy` alternative
### Description

https://docs.cupy.dev/en/stable/reference/array_api.html recommends the use of the experimental `cupy.array_api` submodule as a mirror to the `numpy.array_api` experimental module.

In NumPy, the `numpy.array_api` experimental submodule introduced in 1.6 (or 1.5, I am not sure) has been removed in 2.0 as the top-level `numpy` module itself is expected to be array API compliant.

In scikit-learn, we test against both `cupy.array_api` and `cupy` directly via the `array_api_compat.cupy` wrapper. However `cupy.array_api` is not fully compliant with the latest version of the standard and we might need to invest efforts in maintaining our own wrapper (for instance to add the `clip` function). However we would rather not maintain a compat wrapper for something that is meant go away in the future.

### Idea or request for content

If cupy is to mirror numpy, then I would also expect the `cupy.array_api` submodule to go away in the future. If this is the case, I think it would be worth updating the doc to state this explicitly and instead point to `array_api_compat.cupy` as a pragmatic way for early array API users / consuming libraries to address the remaining compliance issues with the top level `cupy` module.

### PR DESCRIPTION
**Pull request by kmaehashi**
context: https://github.com/cupy/cupy/issues/8470#issuecomment-2285383328.

Closes #8470 and #8747.

cc/ @leofang

### CODE REVIEW

File: docs/source/reference/array_api.rst [hunk 1]
```diff
@@ -1,6 +1,11 @@
 Python Array API Support
 ========================
 
+.. note::
+   ``cupy.array_api`` will be removed in CuPy v14 because its NumPy counterpart ``numpy.array_api`` has been removed.
+   The root module ``cupy.*`` is now compatible with the Array API specification as it mirrors the NumPy v2 API.
```
Comment (Saransh-cpp) [inline docs/source/reference/array_api.rst:6]:
```suggestion
   The root module ``cupy.*`` is now compatible with the Array API specification as it mirrors the NumPy v2.1 API.
```

File: docs/source/reference/array_api_array.rst [hunk 2]
```diff
@@ -1,6 +1,11 @@
 Array API Compliant Object
 ==========================
 
+.. note::
+   ``cupy.array_api`` will be removed in CuPy v14 because its NumPy counterpart ``numpy.array_api`` has been removed.
+   The root module ``cupy.*`` is now compatible with the Array API specification as it mirrors the NumPy v2 API.
```
Comment (Saransh-cpp) [inline docs/source/reference/array_api_array.rst:6]:
```suggestion
   The root module ``cupy.*`` is now compatible with the Array API specification as it mirrors the NumPy v2.1 API.
```

**Review by Saransh-cpp — COMMENTED** [review]
Thanks for adding this in! I am not aware of the features going into CuPy `v14`, but should this say `v2.1`? NumPy started supporting Array API `2023.12` from `v2.1`. Please ignore if `v2` is the right version here.

**Comment by leofang** [comment]
No, supporting array API has been one of the goals of NumPy 2(.0), and it's been delivered
https://numpy.org/doc/stable/release/2.0.0-notes.html#highlights

**Review by asi1024 — APPROVED** [review]

**Pull request merged**